### PR TITLE
Add systemd unit to trigger continous deployment from devel:openQA

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/105379

---

Deployed on openqaworker7, see https://progress.opensuse.org/issues/105379#note-8

### CODE REVIEW

**Review by kalikiana — APPROVED** [review]
Nice. Quiter clean

File: script/openqa-continuous-update [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+usage() {
+    cat << EOF
+Usage: openqa-continous-update
+Trigger automatic system upgrade and reboot if devel:openQA packages are stable
```
Comment (okurz) [inline script/openqa-continuous-update:8]: copy-paste-fail

File: systemd/opensuse/openqa-continuous-update.timer [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Trigger of continuous openQA deployments, see https://progress.opensuse.org/issues/105379
+
+[Timer]
+OnBootSec=10min
```
Comment (okurz) [inline systemd/opensuse/openqa-continuous-update.timer:5]: https://progress.opensuse.org/issues/105379#Suggestions says 5 minutes. I would be ok to accept 5 minutes as continuous, but 10 minutes? I don't know ;) But seriously, I would prefer 5 minutes. Any good reason against?

**Review by okurz — CHANGES_REQUESTED** [review]

File: systemd/opensuse/openqa-continuous-update.timer (hunk 2 above)
Comment (Martchus) [reply systemd/opensuse/openqa-continuous-update.timer:5]: No, I have nothing against 5 minutes. It'll usually just exiting very quickly anyways. I suppose if an update takes longer than 5 minutes the systemd timer will be smart enough to handle it.

Comment (okurz) [reply systemd/opensuse/openqa-continuous-update.timer:5]: Zypper would fail when trying to get the lock

File: script/openqa-continuous-update [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+while true; do
+  case "$1" in
+    -h | --help ) usage 0; shift ;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+repo_name=${DEVEL_OPENQA_REPOSITORY:-devel_openQA}
+if zypper -n ref -r "$repo_name" | grep -q 'is up to date' && "$(dirname "${BASH_SOURCE[0]}")"/openqa-check-devel-repo; then
```
Comment (okurz) [inline script/openqa-continuous-update:27]: Why do you include `openqa-check-devel-repo` in here as well? IMHO if zypper says it has updated packages, which is from download.opensuse.org or any configured mirror, then we should not care querying the API of OBS. Can we just leave that out?

File: dist/rpm/openQA.spec [hunk 4]
```diff
@@ -257,13 +257,24 @@ Covering both openQA and also os-autoinst test engine.
...
 and rebooting the system if devel:openQA packages are stable.
 
+%package continuous-update
+Summary:        Continuously update packages from devel:openQA
+Group:          Development/Tools/Other
+Requires:       %{name}-common
+Requires:       curl
+
+%description continuous-update
+Use this package to install and enable a systemd service for continuously upgrading
```
Comment (okurz) [inline dist/rpm/openQA.spec:275]: We should describe here the relation to "auto-update".

As auto-update does a `zypper dup` without repo restrictions and then even triggers a full system reboot they are complementary, not alternatives.

Please also extend the documentation. See
https://github.com/os-autoinst/openQA/blob/master/docs/Installing.asciidoc#automatic-system-upgrades-and-reboots-of-openqa-hosts= about the according auto-update section where you could extend.

**Review by okurz — CHANGES_REQUESTED** [review]

File: dist/rpm/openQA.spec (hunk 4 above)
Comment (Martchus) [reply dist/rpm/openQA.spec:275]: > they are complementary, not alternatives

Yes. I also have both mechanisms still in place on openqaworker7 and also don't see it as part of the initial ticket to remove the existing auto update.

File: script/openqa-continuous-update (hunk 3 above)
Comment (okurz) [reply script/openqa-continuous-update:27]: hm, ok. It won't be a promise that the packages that are actually downloadable are ok but I guess that makes sense to be aligned with our current development approach.

**Review by okurz — APPROVED** [review]

**Pull request merged**